hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/utils/AzcopyToolHelper.java

```diff
@@ -0,0 +1,245 @@
+        renameProcess.waitFor();
+
+        // Remove the downloaded tarball and azcopy folder
+        String cleanupCmd = REMOVE_CMD + hadoopAzureDir.getAbsolutePath() + AZCOPY_DOWNLOADED_TAR_NAME;
+        String[] cleanupCmdArr = {AZCOPY_CMD_SHELL, AZCOPY_CMD_OPTION, cleanupCmd};
+        Process cleanupProcess = Runtime.getRuntime().exec(cleanupCmdArr);
+        cleanupProcess.waitFor();
+
+        // Set the execute permission on the azcopy executable
+        String chmodCmd = CHMOD_CMD + azcopyDirPath;
+        String[] chmodCmdArr = {AZCOPY_CMD_SHELL, AZCOPY_CMD_OPTION, chmodCmd};
+        Process chmodProcess = Runtime.getRuntime().exec(chmodCmdArr);
+        chmodProcess.waitFor();
+      }
+    }
+    // Change working directory to the hadoop-azure directory
+    System.setProperty(USER_DIR_SYSTEM_PROPERTY, hadoopAzureDir.getAbsolutePath());
+  }
+
+  private File findHadoopAzureDir(File dir) {
+    if (dir == null) {
+      return null;
+    }
+    File[] files = dir.listFiles();
+    if (files == null) {
+      return null;
+    }
+    for (File file : files) {
+      if (file.isDirectory() && file.getName().equals(HADOOP_AZURE_DIR)) {
+        return file;
+      } else {
+        File hadoopAzureDir = findHadoopAzureDir(file);
+        if (hadoopAzureDir != null) {
+          return hadoopAzureDir;
+        }
+      }
+    }
+    return null;
+  }
+
+  private void createFileOrFolder(String pathFromContainerRoot, boolean isFile) throws Exception {
+    downloadAzcopyExecutableIfNotPresent();
+    String url = "https://" + accountName + FORWARD_SLASH + fileSystemName + FORWARD_SLASH +
+        pathFromContainerRoot;
+
+    if (isFile) {
+      createFileCreationScript(azcopyDirPath, "createFile" + Thread.currentThread().getName() + ".sh", azcopyDirPath, sasToken, url);
+    } else {
+      createFolderCreationScript(azcopyDirPath, "createFolder" + Thread.currentThread().getName() + ".sh", azcopyDirPath, sasToken, url);
+    }
+
+    String path;
+    if (isFile) {
+      path = azcopyDirPath + "/createFile" + Thread.currentThread().getName() + ".sh";
+    } else {
+      path = azcopyDirPath + "/createFolder" + Thread.currentThread().getName() + ".sh";
+    }
+    try {
+      ProcessBuilder pb = new ProcessBuilder(path);
+      Process p = pb.start();
+      // wait for the process to finish
+      int exitCode = p.waitFor();
+    } catch (IOException e) {
+      throw new IOException(e.getMessage());
+    } catch (InterruptedException e) {
+      throw new InterruptedException(e.getMessage());
+    }
+    String cleanupCmd = REMOVE_CMD + path;
+    String[] cleanupCmdArr = {AZCOPY_CMD_SHELL, AZCOPY_CMD_OPTION, cleanupCmd};
+    Process cleanupProcess = Runtime.getRuntime().exec(cleanupCmdArr);
+    cleanupProcess.waitFor();
+  }
+
+  private static void createFileCreationScript(String folderPath, String scriptName, String azcopyPath, String sasToken, String containerName) {
```

Comment:
lets create a shell file only once for CI, and then just use it with giving bash args for the required path.

Triggering bash commands via java is istself time-consuming, would be awesome if we reduce other java based recurring tasks (like create and setExecutable).

Reply:
Thanks for the great suggestion. Changed the design to incorporate this